### PR TITLE
forms: add notification preferences form and handle

### PR DESCRIPTION
**Pull request by rekt-hard**
ui: add notification settings preferences template

:heart: Thank you for your contribution!

### Description

closes https://github.com/inveniosoftware/invenio-communities/issues/864

Add a notification settings model, form and handler.
Add UI template for setting section.

### Checklist

Ticks in all boxes and 🟢 on all GitHub actions status checks are required to merge:

- [ ] I'm aware of the [code of conduct](https://inveniordm.docs.cern.ch/contribute/code-of-conduct/).
- [ ] I've created [logical separate commits](https://inveniordm.docs.cern.ch/develop/best-practices/commits/#commits) and followed the [commit message format](https://inveniordm.docs.cern.ch/develop/best-practices/commits/#commit-message).
- [ ] I've added relevant test cases.
- [ ] I've added relevant documentation.
- [ ] I've marked [translation strings](https://inveniordm.docs.cern.ch/develop/best-practices/i18n/) (for relevant code).
- [ ] I've followed the [CSS/JS](https://inveniordm.docs.cern.ch/develop/best-practices/css-js/) and [React](https://inveniordm.docs.cern.ch/develop/best-practices/react/) guidelines (for relevant code).
- [ ] I've followed the [web accessibility](https://inveniordm.docs.cern.ch/develop/best-practices/accessibility/) guidelines (for relevant code).
- [ ] I've followed the [user interface](https://inveniordm.docs.cern.ch/develop/best-practices/ui/) guidelines (for relevant code).
- [ ] I've identified the [copyright holder(s)](https://inveniordm.docs.cern.ch/contribute/copyright-policy/) and updated copyright headers for touched files (>15 lines contributions).
- [ ] I've NOT included third-party code (copy/pasted source code or new dependencies).

**Third-party code**

If you've added [third-party code (copy/pasted or new dependencies)](https://inveniordm.docs.cern.ch/develop/best-practices/commits/#third-party-codedependencies), please reach out to an [architect](https://github.com/orgs/inveniosoftware/teams/architects/members).

**Reminder**

By using GitHub, you have already agreed to the [GitHub’s Terms of Service](https://help.github.com/articles/github-terms-of-service/#6-contributions-under-repository-license) including that:

1. You license your contribution under the same terms as the current repository’s license.
2. You agree that you have the right to license your contribution under the current repository’s license.

### CODE REVIEW

**Comment by kpsherva** [comment]
could you add a screenshot of where these options are displayed ?

File: invenio_users_resources/templates/semantic-ui/invenio_users_resources/settings/notifications.html [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+Copyright (C) 2023 Graz University of Technology.
+
+Invenio-Users-Resources is free software; you can redistribute it and/or
+modify it under the terms of the MIT License; see LICENSE file for more
+details.
+#}
+
+{% from "invenio_userprofiles/settings/_macros.html" import render_field, form_errors %}
+
+{% set panel_icon = "bell icon" %}
```
Comment (slint) [inline invenio_users_resources/templates/semantic-ui/invenio_users_resources/settings/notifications.html:13]: minor: we can hardcode this directly into the box

**Review by slint — APPROVED** [review]
LGTM! One comment about extensibility of the Jinja template, but not critical

File: invenio_users_resources/templates/semantic-ui/invenio_users_resources/settings/notifications.html [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+
+{%- block settings_content scoped %}
+
+<div class="ui segments">
```
[the remark this answers is not in the document]
Comment (rekt-hard) [reply invenio_users_resources/templates/semantic-ui/invenio_users_resources/settings/notifications.html:17]: This segment is only displaying the general notifications (i.e. the top section [app-rdm pr](https://github.com/inveniosoftware/invenio-app-rdm/pull/2237)
In order to add a new section, the [final template](https://github.com/inveniosoftware/invenio-app-rdm/pull/2237/files#diff-361122ccf9e4fa53e8db7468de882511efef6eb49425ea3c9883cdc8ab67cac2R11-R17) can include another similar partial template.

Or do you mean setting up a base settings section template/macro, which already defines a basic segment? Something like
```
  <div class="ui segment secondary">
    {%- block section_header %}
      <i class="{{ panel_icon }}"></i>
      <strong class="header item">{{ _("Notifications") }}</strong>
    {%- endblock section_header %}
  </div>
  <div class="ui segment">
    {%- block section_content %}
    {%- endblock section_content %}
  </div>
```

File: invenio_users_resources/forms.py [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+        super().populate_obj(user)
+
+
+def handle_notifications_form(form):
+    """Handle notification preferences form."""
+    form.populate_obj(current_user)
+    db.session.add(current_user)
+    current_app.extensions["security"].datastore.commit()
+    # NOTE: Flash message after successful update of notification preferences.
+    flash(_("Notification preferences were updated."), category="success")
```
[the remark this answers is not in the document]
Comment (rekt-hard) [reply invenio_users_resources/forms.py]: moved to [`invenio-app-rdm`](https://github.com/inveniosoftware/invenio-app-rdm/pull/2237/files#diff-67f3e936916949abce77f09667b4603c6b0c696b75f04920fcfdcc145e385300R125-R129)

File: invenio_users_resources/templates/semantic-ui/invenio_users_resources/settings/notifications.html (hunk 2 above)
[the remark this answers is not in the document]
Comment (slint) [reply invenio_users_resources/templates/semantic-ui/invenio_users_resources/settings/notifications.html:15]: Ok, I understood better now, got a bit confused with the Semantic-UI classes :)

In that case indeed, adding a new box/section is as easy as including the partial template with a new form.

**Review by slint — APPROVED** [review]

**Pull request merged**